lib/pmdtester/parsers/options.rb

```diff
@@ -0,0 +1,55 @@
+require 'slop'
+
+module PmdTester
+  class Options
+
+    VERSION = '1.0.0'
+
+    attr_reader :local_git_repo
+    attr_reader :base_branch
+    attr_reader :patch_branch
+    attr_reader :base_config
+    attr_reader :patch_config
+    attr_reader :config
+    attr_reader :project_list
+    attr_reader :mode
+
+    def initialize(argv)
+      options = parse(argv)
+      @local_git_repo = options[:r]
+      @base_branch = options[:b]
+      @patch_branch = options[:p]
+      @base_config = options[:bc]
+      @patch_config = options[:pc]
+      @config = options[:c]
+      @project_list = options[:l]
+      @mode = options[:m]
+    end
+
+    private
+
+    def parse(argv)
+      Slop.parse argv do |o|
+        o.string '-r', '--local-git-repo', 'path to the local PMD repository'
+        o.string '-b', '--base-branch', 'name of the base branch in local PMD repository'
+        o.string '-p', '--patch-branch', 'name of the patch branch in local PMD repository'
+        o.string '-bc', '--base-config', 'path to the base PMD configuration file'
+        o.string '-pc', '--patch-config', 'path to the patch PMD configuration file'
+        o.string '-c', '--config', 'path to the base and patch PMD configuration file'
+        o.string '-l', '--list-of-projects', 'path to the file which contains the list of standard projects'
+        o.string '-m', '--mode', "the mode of the tool: 'local', 'online' or 'single'\n" +
+            "\tsingle: Set this option to 'single' if your patch branch contains changes for any option that can't work on master/base branch\n" +
```

Comment:
I'm afraid, I don't understand this. Maybe we can skip this mode "single", until we need it? So, we'll only implement the features, that we need, but nothing more...

Reply:
Single mode is for new rules or new features, and we need to test the performance of new rules or new features on standard projects. The old version doesn't need to be involved in this case.

Reply:
I agree with Andreas here... we should start with as little flags as we actually need, and add more as we go and see a use case arise. I'm even doubtful of the local flag (when I work from home I want to check my changes performance against a baseline... why use a local run when I have a result freely available? I can only think of some kind of "offline mode" as a valid usage for it, and I'd still consider it)

Reply:
When we implement local mode (my prototype is to implement this mode), other modes are easy to implement. The application scenarios of these three modes are predictable. When we integrate tools into CI, we certainly use online mode; when developers want to compare specific two branches/tags of PMD("result freely available" you mentioned is not present in this case), local mode is a good choice.